### PR TITLE
#4187 - Change Streams fullDocumentBeforeChange support

### PR DESCRIPTION
**Pull request by kufd**
Hi,

Here i tried to enable Change Streams fullDocumentBeforeChange feature

### CODE REVIEW

File: spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/ChangeStreamEvent.java [hunk 1]
```diff
@@ -157,17 +165,38 @@ public T getBody() {
...
+	}
+
+	@Nullable
+	public T getBodyBeforeChange() {
+
+		if (raw == null) {
+			return null;
+		}
+
+		Document fullDocumentBeforeChange = raw.getFullDocumentBeforeChange();
```
Comment (mp911de) [inline spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/ChangeStreamEvent.java:178]: Nit: Could be combined with `if(raw == null || raw.getFullDocumentBeforeChange() == null)`

Comment (kufd) [reply spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/ChangeStreamEvent.java]: Fixed

**Comment by christophstrobl** [comment]
LGTM 👍

**Comment by christophstrobl** [comment]
Thank you very much @kufd! Merged to _main_ branch via (aa35aae).

**Pull request closed without merging**